### PR TITLE
Group properties

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Grouped the properties based on the names of https://codeguide.co/#css-declaration-order, no order is changed.

I added the `opacity` property to the Visual group (while it's in the misc group in the codeguide).

Didn't group these properties since I think we should move these to the `misc` group:
https://github.com/twbs/stylelint-config-twbs-bootstrap/blob/31d50c2b7bd43e219e2a24e8248e800986257be2/css/index.js#L167-L169

The benefit of grouping is we get better error reporting, for example:
```
 58:5  ✖  Expected "padding-top" to come before "margin" in group "Box Model"      order/properties-order
 61:5  ✖  Expected "line-height" to come before "color" in group "Typography"      order/properties-order
```

### CODE REVIEW

**Comment by XhmikosR** [comment]
I'm defer this to @mdo. Just let me know if this gets merged if we should make it a major/minor bump :)

**Comment by XhmikosR** [comment]
LGTM AFAICT. Also now that I think about it, this could be just a patch bump, right?

**Comment by MartijnCuppens** [comment]
> LGTM AFAICT. Also now that I think about it, this could be just a patch bump, right?

Yup, no order has changed in the current implementation. If we're going to add 
https://github.com/twbs/stylelint-config-twbs-bootstrap/blob/31d50c2b7bd43e219e2a24e8248e800986257be2/css/index.js#L167-L169
to the misc group we will have a BC though.

File: css/index.js [hunk 1]
```diff
@@ -36,215 +36,240 @@ module.exports = {
...
+          'outline-style',
+          'outline-color',
+          'outline-offset',
+          'box-shadow',
+          'opacity',
+          '-ms-interpolation-mode',
+        ]
+      },
+      {
+        groupName: 'Misc',
```
Comment (adamkudrna) [inline css/index.js:246]: I think even the misc. rules can be split into groups. Over the years (and like many other developers), I've also ended up with an ordering convention similar to @mdo's. But I've never been a fan of group names like "CSS3", "Misc", or "Other", as it suggests something can be improved.

Property groups I currently use are:

1. All properties (`all`)
2. Content (`content`)
3. Position
4. General appearance (`appearance`)
4. Box model
5. Typography
6. Decorations
7. Effects and transforms
8. Interactions
9. Transitions and animations

IMO, it:

- better reflects what's important for rendering the element, putting `content`, `all` or `appearance` up in the list,
- groups "everything"* so there is no "misc"-like group.

\* it's hard to keep the list complete

What can be improved:

- consider reducing/merging single-property groups (1, 2, 4) — these are really special and essential rules, but maybe not worth having their own groups;
- add a group with print related properties like `page-break-after` etc.

What do you guys think?

Comment (adamkudrna) [reply css/index.js:246]: BTW, this is interesting:

> A Stylelint config that sorts CSS properties the way Recess did and Bootstrap did/does.

https://github.com/stormwarning/stylelint-config-recess-order

**Pull request closed without merging**